File: 프로그래머스/1/12940. 최대공약수와 최소공배수/최대공약수와 최소공배수.cpp
#include <string>
#include <vector>

using namespace std;

vector<int> solution(int n, int m) {
    vector<int> answer;
    int r = 0 ;
    int gcd = m ;
    int x = n;
     while(x % gcd != 0) {
        r = x % gcd;
        x = gcd;
        gcd = r;
    }
    int gbd = (n*m)/gcd;
    answer.push_back(gcd);
    answer.push_back(gbd);
    return answer;
}

File: 프로그래머스/1/12940. 최대공약수와 최소공배수/최대공약수와 최소공배수.java
class Solution {
    public int[] solution(int n, int m) {
        return new int[]{gcd(n,m),n*m/gcd(n,m)};
    }
    int gcd(int a, int b){
        int tmp,k;
        
        if(b>a){
            tmp = b;
            b = a;
            a = tmp;
        }
        
        while(b!=0){
            k = a%b;
            a = b;
            b = k;
        }
        return a;
    }
}